src/portable/portable.c

```diff
@@ -264,16 +276,29 @@ static int extract_now(
                         }
 
                         if (socket_fd >= 0) {
+#if HAVE_SELINUX
+                                /* The units will be copied on the host's filesystem, so if they had a SELinux label
+                                 * we have to preserve it. Copy it out so that it can be applied later. */
+                                _cleanup_freecon_ char *con = NULL;
+
+                                r = fgetfilecon_raw(fd, &con);
+                                if (r < 0 && errno != ENODATA)
+                                        log_debug_errno(errno, "Failed to get SELinux file context from '%s', ignoring: %m", de->d_name);
+#endif
```

Comment:
i'd minimize the `#ifdef`fery i.e. move the declaration of `_cleanup_freecon_ char *con = NULL;` outisde of the ifdef block. That way you can drop the ifdeffery further down entirely

Reply:
That's how it was originally, but freecon comes from libselinux so it failed to build without it

Reply:
oh, i see.

well, maybe make _cleanup_freecon_ a nop macro in that case. i mean, if you don't link against libselinux, then you cannot possibly allocate a context label, and thus you cannot possible free it, and thus it's totally safe to just nop it in that case.